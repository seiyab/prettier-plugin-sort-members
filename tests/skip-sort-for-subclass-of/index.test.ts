import { describe, test, expect } from "bun:test";
import { readdir, readFile } from "node:fs/promises";
import { join } from "node:path";
import { format } from "prettier";

const plugins = ["./index.ts"];

const dir = join(import.meta.dir, "testdata");
const files = await readdir(dir);

describe("react/sort-comp", () => {
	const supers = [[[]], [["React.Component"]], [["Component"]], [["Promise"]]];
	describe.each(supers)("%j", async (sp) => {
		test.each(files.filter((f) => f.startsWith("react")))(
			"%s",
			async (name) => {
				const opts = {
					skipSortForSubclassOf: sp,
					sortMembersAlphabetically: true,
				};
				const path = join(dir, name);
				const code = await readFile(path, "utf-8");
				const result = await format(code, {
					...opts,
					filepath: path,
					plugins,
				});
				expect(result).toMatchSnapshot();
			},
		);

		test.each(files)("parser agnostic: %s", async (name) => {
			const path = join(dir, name);
			const opts = {
				skipSortForSubclassOf: sp,
				sortMembersAlphabetically: true,
				filepath: path,
				plugins,
			};
			const code = await readFile(path, "utf-8");
			const result1 = await format(code, {
				...opts,
				parser: "typescript",
			});
			const result2 = await format(code, {
				...opts,
				parser: "babel-ts",
			});
			expect(result1).toBe(result2);
		});
	});
});
