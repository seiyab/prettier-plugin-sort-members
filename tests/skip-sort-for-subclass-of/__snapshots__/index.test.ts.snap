// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`react/sort-comp [] react-1.jsx 1`] = `
"import * as React from "react";

class MyComponent extends React.Component {
  state = {
    serverUrl: "https://localhost:1234",
  };

  componentDidMount() {
    this.setupConnection();
  }

  componentDidUpdate(prevProps, prevState) {
    if (
      this.props.roomId !== prevProps.roomId ||
      this.state.serverUrl !== prevState.serverUrl
    ) {
      this.destroyConnection();
      this.setupConnection();
    }
  }

  onClick() {}
  render() {
    return null;
  }
}
"
`;

exports[`react/sort-comp ["React.Component"] react-1.jsx 1`] = `
"import * as React from "react";

class MyComponent extends React.Component {
  state = {
    serverUrl: "https://localhost:1234",
  };

  componentDidMount() {
    this.setupConnection();
  }

  componentDidUpdate(prevProps, prevState) {
    if (
      this.props.roomId !== prevProps.roomId ||
      this.state.serverUrl !== prevState.serverUrl
    ) {
      this.destroyConnection();
      this.setupConnection();
    }
  }

  onClick() {}
  render() {
    return null;
  }
}
"
`;

exports[`react/sort-comp ["Promise"] react-1.jsx 1`] = `
"import * as React from "react";

class MyComponent extends React.Component {
  state = {
    serverUrl: "https://localhost:1234",
  };

  componentDidMount() {
    this.setupConnection();
  }

  componentDidUpdate(prevProps, prevState) {
    if (
      this.props.roomId !== prevProps.roomId ||
      this.state.serverUrl !== prevState.serverUrl
    ) {
      this.destroyConnection();
      this.setupConnection();
    }
  }

  onClick() {}
  render() {
    return null;
  }
}
"
`;
